src/formElementPlugins/listBuilder/src/ApplicationView.tsx

```diff
@@ -0,0 +1,347 @@
+import React, { useEffect, useState } from 'react'
+import {
+  Button,
+  Modal,
+  Form,
+  Segment,
+  Table,
+  TableHeaderCell,
+  TableCell,
+  Card,
+  Icon,
+  Label,
+} from 'semantic-ui-react'
+import { ApplicationViewProps } from '../../types'
+import { ResponseFull } from '../../../utils/types'
+import { TemplateElement, TemplateElementCategory } from '../../../utils/generated/graphql'
+import ApplicationViewWrapper from '../../ApplicationViewWrapper'
+import strings from '../constants'
+
+enum DisplayType {
+  CARDS = 'cards',
+  TABLE = 'table',
+}
+
+interface InputResponseField {
+  id?: number
+  code?: string
+  isValid?: boolean
+  title?: string | null | undefined
+  value: ResponseFull
+}
+
+type ListItem = { [code: string]: InputResponseField }
+
+const ApplicationView: React.FC<ApplicationViewProps> = ({
+  element,
+  parameters,
+  onSave,
+  onUpdate,
+  Markdown,
+  initialValue,
+  currentResponse,
+  applicationData,
+  allResponses,
+}) => {
+  const { isEditable } = element
+  const {
+    label,
+    description,
+    createModalButtonText = strings.BUTTON_LAUNCH_MODAL,
+    modalText,
+    addButtonText = strings.BUTTON_ADD,
+    updateButtonText = strings.BUTTON_UPDATE,
+    deleteItemText = strings.BUTTON_DELETE,
+    inputFields,
+    displayFormat = getDefaultDisplayFormat(inputFields),
+    displayType = DisplayType.CARDS,
+  } = parameters
+
+  const [currentInputResponses, setCurrentInputResponses] = useState<InputResponseField[]>(
+    resetCurrentResponses(inputFields)
+  )
+
+  const [listItems, setListItems] = useState<ListItem[]>(initialValue?.list ?? [])
+  const [selectedListItem, setSelectedListItem] = useState<number | null>(null)
+  const [open, setOpen] = useState(false)
+  const [inputError, setInputError] = useState(false)
+
+  useEffect(() => {
+    onSave({
+      text: 'Not required', // Text value never displayed to user
```

Comment:
hmm this is what we use to deduce if required is empty. I guess custom validation within this plugin would handle absence of value ? Although i don't think there is an easy way to display isRequired but empty (but only do it in strict mode) if validation is done within the plugin

Reply:
This plugin doesn't actually have a way to enforce "isRequired" yet. Can come up with one (i.e. using presence or absence of `.text`), or could just specify a validation criteria such that `listItems.length > 0` ?

Reply:
btw suggestion is to have at least something there or undefined if itemList.length === 0

Reply:
Cool, done as part of 1aeae727